js/piwik.js

```diff
@@ -2424,12 +2424,27 @@ if (typeof Piwik !== 'object') {
              * Send image request to Piwik server using GET.
              * The infamous web bug (or beacon) is a transparent, single pixel (1x1) image
              */
-            function getImage(request, callback) {
+            function getImage(request, callbacks) {
+
+                callbacks = callbacks || {
+                    success: function() {},
+                    error: function() {}
+                };
+
                 var image = new Image(1, 1);
 
+
                 image.onload = function () {
                     iterator = 0; // To avoid JSLint warning of empty block
-                    if (typeof callback === 'function') { callback(); }
+                    if (typeof callbacks.success === 'function') {
+                        callbacks.success();
+                    }
+                };
+                image.onerror = function () {
+                    iterator = 0; // To avoid JSLint warning of empty block
+                    if (typeof callbacks.success === 'function') {
+                        callbacks.error();
```

Comment:
I think this does break backwards compatibility, does it? In the past eg in `trackLink` users were able to pass a `callback` that was only executed on success. Now it is also executed on error which can break users code.